File: ZOJ/2705.py
#Problem ID: 2705
#Submit Time: 2012-08-30 10:50:38
#Run Time: 20
#Run Memory: 320
#ZOJ User: calvinxiao

import sys

def getline():
    return sys.stdin.readline()

def getint():
    return int(sys.stdin.readline())

def getlist():
    return sys.stdin.readline().strip().split()

lst = []
while 1:
    line = getline()
    if not line:
        break
    m, n = [int(num) for num in line.split()]
    getline()
    ans = 0
    temp = 0
    a = b = 1
    while a + b <= m:
        temp += a
        if m % (a + b) == 0:
            div = m / (a + b)
            ans = max(ans, temp * div * n)
            #print a, b, ans
        a, b = b, a + b
    a = b = 1
    temp = 0
    while a + b <= n:
        temp += a
        #print a, b, temp, temp, temp
        if n % (a + b) == 0:
            if n % (a + b) == 0:
                div = n / (a + b)
                ans = max(ans, temp * div * m)
                #print a, b, temp, ans
        a, b = b, a + b
    lst.append(ans)
for i in xrange(len(lst)):
    print lst[i]
    print ''
    

    

File: ZOJ/2705.c
/*
#Problem ID: 2705
#Submit Time: 2012-08-30 12:22:59
#Run Time: 0
#Run Memory: 160
#ZOJ User: calvinxiao
*/

#include<stdio.h>
long long m, n, ans, temp, a, b;
int main()
{
    while(scanf(" %lld %lld", &n, &m) != EOF) {
        ans = 0; temp = 0; a = b = 1;
        while(a + b <= m || a + b <= n) {
            temp += a;
            if( m % (a + b) == 0) {
                if(ans < temp * n * (m / (a + b)))
                    ans = temp * n * (m / (a + b));
            }
            if( n % (a + b) == 0) {
                if(ans < temp * m * (n / (a + b)))
                    ans = temp * m * (n / (a + b));
            }
            b = a + b;
            a = b - a;
        }
        printf("%lld\n\n", ans);
    }
    return 0;
}
